application/common/components/adapters/SagePeopleIdStore.php

```diff
@@ -91,7 +90,7 @@ public function getActiveUser(string $employeeId)
         $activeUsers = $this->getFromIdStore(sprintf(
             "WHERE %s='%s' AND fHCM2__Employment_Status__c='Active'",
             self::PROP_EMPLOYEE_ID,
-            $employeeId
+            preg_replace('/[^\w]/', '', $employeeId)
```

Comment:
This looks like it sufficiently closes the risk of characters in the `$employeeId` breaking the where clause's syntax, thanks. Do you think there's any significant risk that it will end up matching a different employee's ID? The only example I can think of that might happen would be if _some_ Sage People employee IDs included a hyphen, but not all of them. I don't know if that's the case, though.

About the only way I can think of that _might_ improve this code was if we threw an exception of the "sanitized" version of the Employee ID didn't match the original version of the Employee ID... basically an assertion that `$employeeId` only contains allowed characters in it. I defer to you as to whether that's worth the extra code to implement it.